### ISSUE TITLE
Add descriptive help() function

### ISSUE THREAD

**Issue by tdicola** [adafruit/micropython#35]
Currently help() is unimplemented, but we should add a descriptive block of text that explains at least the micropython firmware version.  Ideally it would have something with instructions on getting started, like with the esp8266 and microbit ports, would be nice to add too.  See esp8266 help implementation: https://github.com/adafruit/micropython/blob/master/esp8266/help.c

**Comment by tannewt** [comment]
What should we have the `help()` provide? Ideally it'd be short because it will be stored in flash memory.

**Comment by tannewt** [comment]
Add a link to adafruit.com/micropython and autogenerate lists of available modules and their members.

**Comment by tannewt** [comment]
Fixed in 427d33c706236985dff950382e1c6b896b36b1bd. Example:

```
>>> help()
Welcome to Adafruit MicroPython v1.8.6-183-ge8db23d-dirty!

Please visit learn.adafruit.com/category/micropython for project guides.

Built in modules:
        __main__
        builtins
        micropython
        array
        ucollections
        ustruct
        math
        sys
        gc
        urandom
        framebuf
        microcontroller
        nativeio
        board
        uos
        time
        neopixel_write
        samd
        bitbangio
>>>
```
